src/video/sdl2_v.cpp

```diff
@@ -205,6 +205,20 @@ bool VideoDriver_SDL_Base::ClaimMousePointer()
 	return true;
 }
 
+void VideoDriver_SDL_Base::FixMousePointer(bool fix_at)
+{
+	if (_cursor.fix_at == fix_at) return;
+
+	this->VideoDriver::FixMousePointer(fix_at);
+
+	SDL_SetRelativeMouseMode(fix_at ? SDL_TRUE : SDL_FALSE);
```

Comment:
No clue if it had an impact here, but do know that relative mode captures the mouse and makes it impossible to leave the window. We once tried to switch to relative mode in general, but that was shitty.

Also, SDL in waylaid emulates relative mode, which might have weird side-effects.

I mostly write this as 4 years ago or something I found out to stay far away from relative mode on Linux :D No clue if it still applies or if it applies to the way you implemented this :) So take this piece of wisdom for what it is worth: random knowledge :D

Reply:
Yes, it does affect that, we could perhaps store the "no_mouse_capture" setting so that we don't actually do the capture.

Reply:
no_mouse_capture is now handled, though it does make the logic more complex as the behaviour between capture and no capture is now a bit different.